src/tests/Microsoft.PowerFx.Interpreter.Tests/IRTests.cs

```diff
@@ -64,6 +64,20 @@ public void RecordToRecordAggregateCoercionTest(string expr)
             Assert.IsNotType<ErrorValue>(result);
         }
 
+        [Theory]
+        
+        [InlineData("[{a:0},{a:\"3\",b:2}]", "Table({a:0,b:If(false,0)},{a:3,b:2})")]
+        [InlineData("[{a:1}, {a:true, b:2}]", "Table({a:1,b:If(false,0)},{a:1,b:2})")]
```

Comment:
>[InlineData("[{a:1}, {a:true, b:2}]", "Table({a:1,b:If(false,0)},{a:1,b:2})")] [](http://example.com/codeflow?start=8&length=78)

Can we add negative tests? Ie what happens when field can't be coerced to certain type.

Reply:
Test case added. Results in a compilation error.